lacommunaute/forum/tests.py

```diff
@@ -29,6 +29,20 @@
 remove_perm = get_class("forum_permission.shortcuts", "remove_perm")
 
 
+def createPostLikeVotesInForum(forum, like=False, vote=False):
+    poster = UserFactory()
+    post = PostFactory(topic=create_topic(forum=forum, poster=poster), poster=poster)
+
+    if like:
+        post.topic.likers.add(poster)
+        post.topic.save()
+
+    if vote:
+        TopicPollVoteFactory(poll_option__poll__topic=post.topic, voter=poster)
+
+    return post.topic
```

Comment:
Quelques remarques :
1. Je trouve ça bizarre que la fonction retourne un objet  `Topic` alors que son nom ne le mentionne pas, bien au contraire d'ailleurs.
2. J'ai pas (du tout) le code en tête mais j'ai l'impression que la logique de cette fonction pourrais être mise dans la factory.
  a. Un `factory.Trait` devrais pouvoir remplacer le `if vote: ...`
  b. Pour `like` le premier truc qui me viens en tête c'est d'utiliser un post-generation hooks
  c. Le lien entre `Post`, `Topic`, `User` devrais probablement pouvoir être fait avec des `factory.SubFactory()` et `factory.SelfAttribute()`
3. Le nom de la fonction n'est pas très [PEP 8](https://peps.python.org/pep-0008/#function-and-variable-names) ;).

Reply:
Merci pour le challenge @rsebille, grâce à toi, j'aurai bosser les `factory.Trait` et la `post_generation` presque dans tous les sens :fire: